src/forEachObj.test.ts

```diff
@@ -70,4 +70,22 @@ describe("typing", () => {
       expectTypeOf(value).toEqualTypeOf<boolean | string>();
     });
   });
+
+  test("union of records", () => {
+    forEachObj(
+      {} as Record<number, string> | Record<string, number>,
+      (value, key) => {
+        expectTypeOf(key).toEqualTypeOf<string>();
```

Comment:
wouldn't the key here need to be:
```ts
`${number}` | string
```

Is that being squished by TypeScript because they are both strings?

Reply:
I think it is indeed being squished by TypeScript as the first one is a subset of string. I tried this example below:
```ts
type A = `${number}`
type B = string
type C = A | B // C is of type string
```